pkg/vmextension/status/status.go

```diff
@@ -54,15 +56,25 @@ func (r StatusReport) marshal() ([]byte, error) {
 	return json.MarshalIndent(r, "", "\t")
 }
 
-// Save persists the status message to the specified status folder
-// using the sequence number.
+// Save persists the status message to the specified status folder using the
+// sequence number. The operation consists of writing to a temporary file in the
+// same folder and moving it to the final destination for atomicity.
 func (r StatusReport) Save(statusFolder string, seqNum int) error {
+	suffix := fmt.Sprintf(".tmp.%d", rand.New(rand.NewSource(time.Now().UnixNano())).Int63())
```

Comment:
actually I'm changing this to `ioutil.TempFile(dir, "")` , it has an even more thorough logic https://golang.org/src/io/ioutil/tempfile.go#L47